### DISCUSSION TITLE
Finish time in server testing

### DISCUSSION THREAD

**Discussion by TomohiroOno33** [hsr-project/general#13] [Q&A] [answered]
When testing on a server, is there any way to finish the test within 21 minutes and return the results?

**Comment by yosuke** [comment, marked as answer]
The launch file of the template project is configured with `required="true"` option to launch the main algorithm:
https://github.com/devrt/robocup-at-home-2021-challenge/blob/main/robocup_challenge/launch/run.launch#L2

So if you exit from the main algorithm (by using `sys.exit()` API) simulation will be stopped at that point and return the result.
Monitoring the simulation time by rospy time API is recommended.

We also observe some algorithm does not stop even if the simulation server sends a kill signal to them which may be caused by a memory leak problem inside the algorithm.

**Reply by TomohiroOno33** [reply]
Thank you for the useful information.
I just tested it and confirmed that it works.
